src/Processors/Formats/Impl/ProtobufRowInputFormat.cpp

```diff
@@ -66,21 +83,43 @@ void ProtobufRowInputFormat::syncAfterError()
     reader->endMessage(true);
 }
 
-void registerInputFormatProtobuf(FormatFactory & factory)
-{
-    for (bool with_length_delimiter : {false, true})
-    {
-        factory.registerInputFormat(with_length_delimiter ? "Protobuf" : "ProtobufSingle", [with_length_delimiter](
-            ReadBuffer & buf,
-            const Block & sample,
-            IRowInputFormat::Params params,
-            const FormatSettings & settings)
+void registerInputFormatProtobuf(FormatFactory & factory){
+    /// proton: starts
+    /// for (bool with_length_delimiter : {false, true})
+    /// {
+    ///     factory.registerInputFormat(
+    ///         with_length_delimiter ? "Protobuf" : "ProtobufSingle",
+    ///         [with_length_delimiter](
+    ///             ReadBuffer & buf, const Block & sample, IRowInputFormat::Params params, const FormatSettings & settings) {
+    ///             return std::make_shared<ProtobufRowInputFormat>(
+    ///                 buf, sample, std::move(params), FormatSchemaInfo(settings, "Protobuf", true), with_length_delimiter);
+    ///         });
+    /// }
+
+    factory.registerInputFormat(
+        "ProtobufSingle",
+        [](ReadBuffer & buf, const Block & sample, IRowInputFormat::Params params, const FormatSettings & settings) -> std::shared_ptr<IInputFormat>
         {
-            return std::make_shared<ProtobufRowInputFormat>(buf, sample, std::move(params),
-                FormatSchemaInfo(settings, "Protobuf", true),
-                with_length_delimiter);
+            if (settings.schema.kafka_schema_registry_url.empty())
+                return std::make_shared<ProtobufRowInputFormat>(
+                    buf, sample, std::move(params), FormatSchemaInfo(settings, "Protobuf", true), /*with_length_delimiter=*/false);
+
+            if (!settings.schema.format_schema.empty())
```

Comment:
Since we have done the validation here, do we still need do validation in ProtobufRowInputFormat or  ProtobufConfluentRowInputFormat ? Can we just assert there instead ?